src/node-project.ts

```diff
@@ -714,9 +724,17 @@ export class NodeProject extends GitHubProject {
 
     const autoApproveLabel = (condition: boolean) => (condition && this.autoApprove?.label) ? [this.autoApprove.label] : undefined;
 
-    const defaultDependenciesUpgrade = (options.dependabot ?? false)
-      ? DependenciesUpgradeMechanism.dependabot({ labels: autoApproveLabel(depsAutoApprove) })
-      : DependenciesUpgradeMechanism.githubWorkflow({
+    let ignoresProjen;
+    if (dependabot) {
+      const dependabotConf = this.github?.addDependabot({
+        labels: autoApproveLabel(depsAutoApprove),
+        ...(options.dependabotOptions ?? {}),
+      });
+      ignoresProjen = dependabotConf?.ignoresProjen;
+    }
+
+    if (depsUpgrade) {
+      const upgradeDependencies = new UpgradeDependencies(this, {
         // if projen secret is defined we can also upgrade projen here.
         ignoreProjen: !options.projenUpgradeSecret,
```

Comment:
This is outside of the immediate scope of this PR, but there's currently a small bug where the projen repository itself should be auto-approving PRs, but it is not because the upgrade.yml workflow is using the `GITHUB_TOKEN` workflow instead of `PROJEN_GITHUB_TOKEN`. Adding `projenUpgradeSecret: 'PROJEN_GITHUB_TOKEN'` to the constructor should fix this - but if we do so, then this "ignoreProjen" flag gets flipped and projen will try upgrading itself.

Could you try adding this line and see if that fixes it? :)
```suggestion
      const upgradeDependencies = new UpgradeDependencies(this, {
        // if projen secret is defined we can also upgrade projen here.
        ignoreProjen: !options.projenUpgradeSecret || this.package.packageName === 'projen',
```

Reply:
I dont think this is necessary actually. The upgrade task won't try to upgrade `projen` because its not a dependency of `projen`. The only thing the explicitly upgrades projen is:

https://github.com/projen/projen/blob/epolon/upgrade-sign-off/src/node-project.ts#L754

Which already special cases `projen` itself. We should change `.projenrc` to configure:

```ts
depsUpgrade: {
  workflowOptions: {
    secret: 'PROJEN_UPGRADE_TOKEN`
  }
}
```

And that should do it. 

I'll take of this in a subsequent PR